src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java

```diff
@@ -200,12 +204,12 @@ public SonarRunnerInstallation getSonarRunnerInstallation() {
   }
 
   @Override
-  public void perform(@Nonnull Run<?,?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws InterruptedException, IOException {
+  public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) throws InterruptedException, IOException {
     performInternal(run, workspace, launcher, listener);
   }
 
-  public boolean performInternal(@Nonnull Run<?,?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws IOException, InterruptedException {
-    if (!isSonarInstallationValid(getInstallationName(), listener)) {
+  public boolean performInternal(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.lang.InterruptedException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1160)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "performInternal" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)